lib/date_time_parser/timezone_abbreviations.ex

```diff
@@ -0,0 +1,327 @@
+defmodule DateTimeParser.TimezoneAbbreviations do
+  @moduledoc """
+  Fetch timezone information by name or by abbreviations. This is designed
+  to power parsing strings with timezone abbreviations and offsets in them.
+  """
+
+  {zones, rules} = DateTimeParser.TimezoneParser.parse()
+  @zones zones
+  @rules rules
+
+  @default_offset_preferences %{
+    "+0000" => "Etc/UTC",
+    "+0100" => "Etc/GMT-1",
+    "+0200" => "Etc/GMT-2",
+    "+0300" => "Etc/GMT-3",
+    "+0400" => "Etc/GMT-4",
+    "+0500" => "Etc/GMT-5",
+    "+0530" => "Asia/Kolkata",
+    "+0600" => "Etc/GMT-6",
+    "+0700" => "Etc/GMT-7",
+    "+0800" => "Etc/GMT-8",
+    "+0900" => "Etc/GMT-9",
+    "+0930" => "Australia/Adelaide",
+    "+1000" => "Etc/GMT-10",
+    "+1100" => "Etc/GMT-11",
+    "+1200" => "Etc/GMT-12",
+    "+1300" => "Etc/GMT-13",
+    "+1400" => "Etc/GMT-14",
+    "-0000" => "Etc/UTC",
+    "-0100" => "Etc/GMT+1",
+    "-0200" => "Etc/GMT+2",
+    "-0300" => "Etc/GMT+3",
+    "-0400" => "Etc/GMT+4",
+    "-0500" => "Etc/GMT+5",
+    "-0600" => "Etc/GMT+6",
+    "-0700" => "Etc/GMT+7",
+    "-0800" => "Etc/GMT+8",
+    "-0900" => "Etc/GMT+9",
+    "-1000" => "Etc/GMT+10",
+    "-1100" => "Etc/GMT+11"
+  }
+
+  @default_abbreviation_preferences %{
+    # Preferred GMT where possible, then USA, otherwise most populous.
+    "+01/+00" => "Africa/Casablanca",
+    "+02" => "Etc/GMT-2",
+    "+03" => "Etc/GMT-3",
+    "+04" => "Etc/GMT-4",
+    "+04/+05" => "Asia/Baku",
+    "+05" => "Etc/GMT-5",
+    "+06" => "Etc/GMT-6",
+    "+07" => "Etc/GMT-7",
+    "+08" => "Etc/GMT-8",
+    "+08/+09" => "Asia/Ulaanbaatar",
+    "+09" => "Etc/GMT-9",
+    "+10" => "Etc/GMT-10",
+    "+11" => "Etc/GMT-11",
+    "+11/+12" => "Pacific/Noumea",
+    "+12" => "Etc/GMT-12",
+    "+13" => "Etc/GMT-13",
+    "+13/+14" => "Pacific/Tongatapu",
+    "+14" => "Etc/GMT-14",
+    "-01" => "Etc/GMT+1",
+    "-01/+00" => "Atlantic/Azores",
+    "-02" => "Etc/GMT+2",
+    "-03" => "Etc/GMT+3",
+    "-03/-02" => "America/Buenos_Aires",
+    "-04" => "Etc/GMT+4",
+    "-04/-03" => "America/Santiago",
+    "-05" => "Etc/GMT+5",
+    "-05/-04" => "America/Bogota",
+    "-06/-05" => "Pacific/Galapagos",
+    "-08" => "Etc/GMT+8",
+    "-09" => "Etc/GMT+9",
+    "-10" => "Etc/GMT+10",
+    "-11" => "Etc/GMT+11",
+    # Australia
+    "AWST" => "Australia/Perth",
+    "ACWST" => "Australia/Eucla",
+    "ACT" => "Australia/Adelaide",
+    "ACDT" => "Australia/Adelaide",
+    "ACST" => "Australia/Adelaide",
+    "AET" => "Australia/Sydney",
+    "AEST" => "Australia/Sydney",
+    "AEDT" => "Australia/Sydney",
+    "LHDT" => "Australia/Lord_Howe",
```

Comment:
@joshprice Do these abbreviation -> timezone assumptions make sense to you?

Reply:
Compare with this https://www.australia.gov.au/time-zones-and-daylight-saving

Reply:
perfect thanks.

Reply:
TIL about Eucla which is an "unofficial" timezone https://en.wikipedia.org/wiki/UTC%2B08:45

Reply:
I'd stick with the official timezones

Reply:
It is indeed an official timezone.